### PR TITLE
feat SER-2803 Add Github actions

### PR DESCRIPTION
**Pull request by amydoxym**
## Description
Move build from Azure to Github Actions

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,27 +1,39 @@
...
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+          show-progress: false
 
       - name: Use Node.js 16
```
Comment (Yann-J) [inline .github/workflows/build.yml:38]: Why 16? That seems a bit old... I see the comment saying it's not supported, but the linked issue (https://github.com/actions/setup-node/issues/26) was closed 2 years ago already...

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -33,35 +45,56 @@ jobs:
...
+          registry: oaftech.azurecr.io
+          username: ${{ vars.ACR_PULL_USERNAME }}
+          password: ${{ secrets.ACR_PULL_PASSWORD }}
 
-      - name: Run build
-        continue-on-error: true
-        run: npm run build:prod
+      - name: Build and push an image to container registry
+        uses: docker/build-push-action@v5.1.0
+        if: ${{ env.SHOULD_PUBLISH == 'true'}}
```
Comment (Yann-J) [inline .github/workflows/build.yml:75]: We should always build, since we want to validate that the iage build works.
However we should not always push.

**Review by Yann-J — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (amydoxym) [reply .github/workflows/build.yml]: @Yann-J Updated to 18

File: .github/workflows/build.yml (hunk 2 above)
Comment (amydoxym) [reply .github/workflows/build.yml:78]: Fixed

File: .github/workflows/gitguardian.yaml
Comment (Yann-J) [inline .github/workflows/gitguardian.yaml:1]: I don't think this is required.
I can see it's running globally already on all PRs...

**Review by Yann-J — CHANGES_REQUESTED** [review]

File: .github/workflows/gitguardian.yaml
Comment (amydoxym) [reply .github/workflows/gitguardian.yaml:1]: Removed

**Review by Yann-J — APPROVED** [review]

**Review by gilclefes — APPROVED** [review]
looks good to me

**Pull request merged**